src/server/client/App.js

```diff
@@ -0,0 +1,96 @@
+import React, { useEffect, useState } from 'react';
+import { BrowserRouter as Router, Switch, Route } from 'react-router-dom';
+import { decodeToken } from './helpers/jwt';
+import { Login, Register, Home } from './pages/';
+import authContext from './state/state';
+// import usuarioContexto from './state/state';
+
+export default function App() {
+  const [user, setUser] = useState({ islogged: false });
+
+  const login = (email = '', password = '') => {
+    if (!email || !password) return;
+
+    fetch('http://localhost:3000/api/login', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ email, password })
+    })
+      .then(resp => resp.json())
+      .then(data => {
+        // guarda el token en el localStorage
+        localStorage.setItem('token', data.token);
+        // cambia el estado de logeado a true
+        setUser({ ...decodeToken(data.token), islogged: true });
+      })
+      .catch(console.log);
+  };
+
+  const register = (nombre, email, password1, password2) => {
+    fetch('http://localhost:3000/api/registro', {
+      method: 'POST',
+      headers: {
+        Accept: 'application/json',
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify({ nombre, email, password1, password2 })
+    })
+      .then(resp => resp.json())
+      .then(data => {
+        localStorage.setItem('token', data.token);
+        // si existe  token , pasa a setear el  token, pero antes lo decifra
+        if (decodeToken) setUser({ ...decodeToken(data.token), islogged: true });
+      });
+  };
+  const logout = isLogin => {
+    !isLogin || setUser({ islogged: false });
+  };
+
+  useEffect(() => {
+    const token = localStorage.getItem('token');
+    // console.log(token, 'hola');
+    if (token) {
+      fetch('http://localhost:3000/api/renew', {
+        headers: {
+          token
+        }
+      })
+        .then(res => res.json())
+        .then(data => {
+          if (data.ok) {
+            localStorage.setItem('token', data.token);
+            console.log(decodeToken(token));
+            setUser({ ...decodeToken(token), islogged: true });
+            // login(nombre, uid);
+            // setUser(p => ({
+            //   ...p,
+            //   islogged: true
+            // }));
+          }
+        });
+    }
+  }, []);
+
+  return (
+    <authContext.Provider value={{ user, login, logout, register }}>
+      <Router>
```

Comment:
I didn't mean this should return the Router. This one should receive `children` as props and just render them.

I meant exporting a component... for instance, `AuthProvider` that uses `AuthContext` and returns:

```
<AuthContext.Provider value={...}>
  {props.children}
</AuthContext.Provider>
```


Then, in your original `App`, you could just do

```
<AuthProvider>
  <Router>...</Router>
</AuthProvider>
```

without having all the auth logic in the main component. Code encapsulation is the concept you want to learn here

Reply:
Ahh this one was the server one... this is confusing now